docs/user_guide/fairness_in_machine_learning.rst

```diff
@@ -141,6 +141,103 @@ which groups these values were observed, as well as the difference and ratio
 between the maximum and the minimum values. For more information refer to the
 subpackage :code:`fairlearn.metrics`.
 
+Failure modes encountered when abstracting
+--------------------------------------------------
+
+Machine learning systems used in the real world are inherently sociotechnical
+systems, which include both technologies and social actors. Designers of machine
+learning systems who fail to consider the way in which social contexts and technologies
+are interrelated are at risk of falling into "abstraction traps". In this section,
+we explain what those traps are, and give some suggestions on how you can avoid them.
+
+In "Fairness and Abstraction in Sociotechnical Systems" Selbst et al. [#4]_
+identify failure modes that arise from applying abstraction. They identify
+them as:
+
+* *The Solutionism Trap*
+
+* *The Ripple Effect Trap*
+
+* *The Formalism Trap*
+
+* *The Portability Trap*
+
+* *The Framing Trap*
+
+These traps are inspired by the observation that each of these traps are the
+result of failing to consider the way in with social context and technology
+are interrelated, as well as a deeper understanding of "the social" in order to
+solve problems Ackerman [#5]_.
+
+The Solutionism Trap
+^^^^^^^^^^^^^^^^^^^^
+
+This trap occurs when data scientists assume that the best solution to a problem involves
+technology, and fail to recognize other possible solutions outside of this realm.
+One area where this manifests in is within contexts in which the definition "fairness"
+changes or is dependent on a political context.
+
+Example: consider the problem of internet connectivity in rural communities. An example
+of the solutionism trap is assuming that by using data science to study internet speed in
+a given region, policies and deals can be arranged to reduce the price of internet access.
+However, if there are additional socioeconomic problems within a community, for example lack of
+education, infrastructure, information technology and health services, then an
+algorithmic solution purely focused on internet speed will fail to meaningfully address
+the needs of the community.
+
+The Ripple Effect Trap
+^^^^^^^^^^^^^^^^^^^^^^
+
+Selbst et al. [#4]_ define this as a "failure to understand how the insertion of technology into
+an existing social system changes the behaviors and embedded values of the pre-existing system".
+
+Example 1:
+
+The Formalism Trap
+^^^^^^^^^^^^^^^^^^
+
+This trap occurs when implementing an algorithmic solution failing to account the
+social dimensions associated to a problem. These dimensions include processes involved,
+whether human, political or bureaucratic, as well as cultural and time-bound social
+contexts. Such dimensions can often not be resolved via mathematical means.
+Failure occurs when steps are not taken to eradicate the limitations behind mathematical
```

Comment:
This sentence might be a bit vague for people who are unfamiliar with what these limitations are exactly.

Reply:
agreed, modified quite a bit the formalism section, hopefully it is a bit more clear now